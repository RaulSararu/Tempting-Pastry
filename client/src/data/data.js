const data = {
    breads: [
        {   
            id:"1",
            name:"Assorted 1LB Bread Bowl",
            category:"breads",
            image:"/images/breads/breadbowl.png", 
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"2",
            name:"Assorted 4x4 Ciabatta",
            category:"breads",
            image:"/images/breads/ciabatta.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"3",
            name:"Olive Parmesean Pumperknickle",
            category:"breads",
            image:"/images/breads/olive.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"4",
            name:"Pretzel Rolls",
            category:"breads",
            image:"/images/breads/pretzel.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"5",
            name:"Seven Grain Rosemary",
            category:"breads",
            image:"/images/breads/sevengrain.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        }
    ],
    breakfast : [
        {   
            id:"6",
            name:"Apple Cheese Spinwheel",
            category:"",
            image:"/images/breakfast/apple.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"7",
            name:"Cheese Croissant",
            category:"",
            image:"/images/breakfast/cheese.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"8",
            name:"Chocolate Croissant",
            category:"",
            image:"/images/breakfast/chocolate.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"9",
            name:"Raspberry Croissant",
            category:"",
            image:"/images/breakfast/raspberry.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"10",
            name:"White Chocolate Macadamia",
            category:"",
            image:"/images/breakfast/white.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        }
    ],
    cakes: [
        {   
            id:"11",
            name:"Chocolate Mousse",
            category:"",
            image:"/images/cakes/mousse.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"12",
            name:"LA Cheese Cake",
            category:"",
            image:"/images/cakes/cheese.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"13",
            name:"Mango Mousse",
            category:"",
            image:"/images/cakes/mango.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"14",
            name:"Pina Colada",
            category:"",
            image:"/images/cakes/pina.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"15",
            name:"Red Velvet",
            category:"",
            image:"/images/cakes/redvelvet.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"16",
            name:"Tiramisu",
            category:"",
            image:"/images/cakes/tiramisu.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        }
    ],
    desserts: [
        {   
            id:"17",
            name:"Cannoli",
            category:"",
            image:"/images/desserts/cannoli.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"18",
            name:"Cappuccino Mousse",
            category:"",
            image:"/images/desserts/cappuccino.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"19",
            name:"Caramel Flan",
            category:"",
            image:"/images/desserts/caramel.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"20",
            name:"Chocolate Eclair",
            category:"",
            image:"/images/desserts/eclair.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        },
        {   
            id:"21",
            name:"Chocolate Grand Marnier",
            category:"",
            image:"/images/desserts/grandmarnier.png",
            price:20,
            countInStock:20,
            rating:4.5,
            numReviews:10,
            description:"",
        }
    ]
}

export default data