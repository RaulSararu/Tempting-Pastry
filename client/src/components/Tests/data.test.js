import data from "../../data/data";

describe("testing data", () => {
  test("data is correct", () => {
    expect(data).toMatchSnapshot();
    expect(1).toEqual(1);
  });

  test("check data length", () => {
    expect(data.breads).toHaveLength(10);
    expect(data.breakfast).toHaveLength(10);
    expect(data.cakes).toHaveLength(11);
    expect(data.desserts).toHaveLength(11);
  });

  test("check data breads name", () => {
    expect(data.breads.map((item) => item.name)).toEqual([
      "Assorted 1LB Bread Bowl",
      "Assorted 4x4 Ciabatta",
      "Olive Parmesean Pumperknickle",
      "Pretzel Rolls",
      "Seven Grain Rosemary",
      "Dozen of Assorted Artisan Rolls",
      "Dozen of Assorted Dinner Rolls",
      "Assorted French Baguette",
      "Foccacia 4x4",
      "Kaiser Rolls",
    ]);
  });

  test("check data breakfast id", () => {
    expect(data.breakfast.map((item) => item.id)).toEqual([
      "11",
      "12",
      "13",
      "14",
      "15",
      "16",
      "17",
      "18",
      "19",
      "20",
    ]);
  });

  test("check data cakes price", () => {
    expect(data.cakes.map((item) => item.price)).toEqual([
      20, 21, 20, 25, 23, 23, 25, 20, 20, 20, 20,
    ]);
  });

  test("check data desserts images", () => {
    expect(data.desserts.map((item) => item.image)).toEqual([
      "/images/desserts/cannoli.png",
      "/images/desserts/cappuccino.png",
      "/images/desserts/caramel.png",
      "/images/desserts/eclair.png",
      "/images/desserts/grandmarnier.png",
      "/images/desserts/mix_fruit_florentine.png",
      "/images/desserts/black_forest.png",
      "/images/desserts/cappuccino_eclair.png",
      "/images/desserts/carrot_cake.png",
      "/images/desserts/chocolate_suicide.png",
      "/images/desserts/mango_mousse.png",
    ]);
  });

  for (let i = 0; i < data.breads.length; i += 1) {
    test(`data[${i}] should have proprieties (name, id, category, price, description, rating, countInStock, image, numReviews )`, () => {
      expect(data.breads[i]).toHaveProperty("id");
      expect(data.breads[i]).toHaveProperty("name");
      expect(data.breads[i]).toHaveProperty("category");
      expect(data.breads[i]).toHaveProperty("price");
      expect(data.breads[i]).toHaveProperty("description");
      expect(data.breads[i]).toHaveProperty("rating");
      expect(data.breads[i]).toHaveProperty("countInStock");
      expect(data.breads[i]).toHaveProperty("image");
      expect(data.breads[i]).toHaveProperty("numReviews");
    });
  }

  for (let i = 0; i < data.cakes.length; i += 1) {
    test(`data[${i}] should have proprieties (name, id, category, price, description, rating, countInStock, image, numReviews )`, () => {
      expect(data.cakes[i]).toHaveProperty("id");
      expect(data.cakes[i]).toHaveProperty("name");
      expect(data.cakes[i]).toHaveProperty("category");
      expect(data.cakes[i]).toHaveProperty("price");
      expect(data.cakes[i]).toHaveProperty("description");
      expect(data.cakes[i]).toHaveProperty("rating");
      expect(data.cakes[i]).toHaveProperty("countInStock");
      expect(data.cakes[i]).toHaveProperty("image");
      expect(data.cakes[i]).toHaveProperty("numReviews");
    });
  }

  for (let i = 0; i < data.breakfast.length; i += 1) {
    test(`data[${i}] should have proprieties (name, id, category, price, description, rating, countInStock, image, numReviews )`, () => {
      expect(data.breakfast[i]).toHaveProperty("id");
      expect(data.breakfast[i]).toHaveProperty("name");
      expect(data.breakfast[i]).toHaveProperty("category");
      expect(data.breakfast[i]).toHaveProperty("price");
      expect(data.breakfast[i]).toHaveProperty("description");
      expect(data.breakfast[i]).toHaveProperty("rating");
      expect(data.breakfast[i]).toHaveProperty("countInStock");
      expect(data.breakfast[i]).toHaveProperty("image");
      expect(data.breakfast[i]).toHaveProperty("numReviews");
    });
  }

  for (let i = 0; i < data.desserts.length; i += 1) {
    test(`data[${i}] should have proprieties (name, id, category, price, description, rating, countInStock, image, numReviews )`, () => {
      expect(data.desserts[i]).toHaveProperty("id");
      expect(data.desserts[i]).toHaveProperty("name");
      expect(data.desserts[i]).toHaveProperty("category");
      expect(data.desserts[i]).toHaveProperty("price");
      expect(data.desserts[i]).toHaveProperty("description");
      expect(data.desserts[i]).toHaveProperty("rating");
      expect(data.desserts[i]).toHaveProperty("countInStock");
      expect(data.desserts[i]).toHaveProperty("image");
      expect(data.desserts[i]).toHaveProperty("numReviews");
    });
  }

  test("mock return value of a function one time", () => {
    const mock = jest.fn(() => "I am a mock function");
    expect(mock("Calling mock function")).toBe("I am a mock function");
    expect(mock).toHaveBeenCalledWith("Calling mock function");
  });

  test("spying using original implementation", () => {
    const bread = {
      name: (n) => `Bread name: ${n}`,
    };
    const spy = jest.spyOn(bread, "name");
    expect(bread.name("Assorted 1LB Bread Bowl")).toBe(
      "Bread name: Assorted 1LB Bread Bowl"
    );
    expect(spy).toHaveBeenCalledWith("Assorted 1LB Bread Bowl");
  });

  test("spying using mockImplementation", () => {
    const bread = {
      name: (n) => `Bread name: ${n}`,
    };
    const spy = jest.spyOn(bread, "name");
    spy.mockImplementation((n) => "Crazy bread");
    expect(bread.name("Assorted 1LB Bread Bowl")).toBe("Crazy bread");
    spy.mockRestore();
    expect(bread.name("Assorted 1LB Bread Bowl")).toBe(
      "Bread name: Assorted 1LB Bread Bowl"
    );
  });

  test("bread returns the last name of the pizza", () => {
    const bread1 = data.breads[0];
    const bread2 = data.breads[1];
    const bread3 = data.breads[2];
    const bread = jest.fn((currentBread) => currentBread.name);

    bread(bread1);
    bread(bread2);
    bread(bread3);

    expect(bread).toHaveLastReturnedWith("Olive Parmesean Pumperknickle");
  });

  test("bread data has Olive Parmesean Pumperknickle and matches as an object", () => {
    const oliveParmeseanPumperknickle = {
      id: "3",
      name: "Olive Parmesean Pumperknickle",
      category: "breads",
      image: "/images/breads/olive.png",
      price: 4,
      countInStock: 20,
      rating: 4.5,
      numReviews: 10,
      description: "",
    };
    expect(data.breads[2]).toMatchObject(oliveParmeseanPumperknickle);
  });

  test("expect a promise to resolve", async () => {
    const user = {
      getFullName: jest.fn(() => Promise.resolve("Raul")),
    };
    await expect(user.getFullName("Raul")).resolves.toBe("Raul");
  });

  test("expect a promise to reject", async () => {
    const user = {
      getFullName: jest.fn(() =>
        Promise.reject(new Error("Something went wrong"))
      ),
    };
    await expect(user.getFullName("Raul")).rejects.toThrow(
      "Something went wrong"
    );
  });
});
