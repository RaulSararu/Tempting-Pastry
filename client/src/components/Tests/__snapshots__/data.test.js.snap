// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing data data is correct 1`] = `
Object {
  "breads": Array [
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "1",
      "image": "/images/breads/breadbowl.png",
      "name": "Assorted 1LB Bread Bowl",
      "numReviews": 10,
      "price": 4,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "2",
      "image": "/images/breads/ciabatta.png",
      "name": "Assorted 4x4 Ciabatta",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "3",
      "image": "/images/breads/olive.png",
      "name": "Olive Parmesean Pumperknickle",
      "numReviews": 10,
      "price": 4,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "4",
      "image": "/images/breads/pretzel.png",
      "name": "Pretzel Rolls",
      "numReviews": 10,
      "price": 4,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "5",
      "image": "/images/breads/sevengrain.png",
      "name": "Seven Grain Rosemary",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "6",
      "image": "/images/breads/assorted_artisan_rolls.png",
      "name": "Dozen of Assorted Artisan Rolls",
      "numReviews": 10,
      "price": 4,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "7",
      "image": "/images/breads/assorted_dinner_rolls.png",
      "name": "Dozen of Assorted Dinner Rolls",
      "numReviews": 10,
      "price": 4,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "8",
      "image": "/images/breads/assorted_french_baguette.png",
      "name": "Assorted French Baguette",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "9",
      "image": "/images/breads/assorted_sandwich_foccacia.png",
      "name": "Foccacia 4x4",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "breads",
      "countInStock": 20,
      "description": "",
      "id": "10",
      "image": "/images/breads/kaiser_rolls.png",
      "name": "Kaiser Rolls",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
  ],
  "breakfast": Array [
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "11",
      "image": "/images/breakfast/apple.png",
      "name": "Apple Cheese Spinwheel",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "12",
      "image": "/images/breakfast/cheese.png",
      "name": "Cheese Croissant",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "13",
      "image": "/images/breakfast/chocolate.png",
      "name": "Chocolate Croissant",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "14",
      "image": "/images/breakfast/raspberry.png",
      "name": "Raspberry Croissant",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "15",
      "image": "/images/breakfast/white.png",
      "name": "White Chocolate Macadamia",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "16",
      "image": "/images/breakfast/almond.png",
      "name": "Almond Croissant",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "17",
      "image": "/images/breakfast/apple_normandie.png",
      "name": "Apple Normandie Danish",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "18",
      "image": "/images/breakfast/cherry.png",
      "name": "Cherry Danish",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "19",
      "image": "/images/breakfast/cinnamon_twist.png",
      "name": "Cinnamon Twist",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "20",
      "image": "/images/breakfast/raspberry.png",
      "name": "Ham and Cheese Croissant",
      "numReviews": 10,
      "price": 2,
      "rating": 4.5,
    },
  ],
  "cakes": Array [
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "21",
      "image": "/images/cakes/mousse.png",
      "name": "Chocolate Mousse",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "22",
      "image": "/images/cakes/cheese.png",
      "name": "La Cheese Cake",
      "numReviews": 10,
      "price": 21,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "23",
      "image": "/images/cakes/mango.png",
      "name": "Mango Mousse",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "24",
      "image": "/images/cakes/pina.png",
      "name": "Pina Colada",
      "numReviews": 10,
      "price": 25,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "25",
      "image": "/images/cakes/redvelvet.png",
      "name": "Red Velvet",
      "numReviews": 10,
      "price": 23,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "26",
      "image": "/images/cakes/tiramisu.png",
      "name": "Tiramisu",
      "numReviews": 10,
      "price": 23,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "27",
      "image": "/images/cakes/carrot_retail.png",
      "name": "Carrot Cake",
      "numReviews": 10,
      "price": 25,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "28",
      "image": "/images/cakes/chocolate_suicide_retail.png",
      "name": "Chocolate Suicide",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "29",
      "image": "/images/cakes/german_chocolate_retail.png",
      "name": "German Chocolate Cake",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "30",
      "image": "/images/cakes/pina_colada_retail.png",
      "name": "Pi√±a Colada",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "31",
      "image": "/images/cakes/tres_leches_retail.png",
      "name": "Tres Leches",
      "numReviews": 10,
      "price": 20,
      "rating": 4.5,
    },
  ],
  "desserts": Array [
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "32",
      "image": "/images/desserts/cannoli.png",
      "name": "Cannoli",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "33",
      "image": "/images/desserts/cappuccino.png",
      "name": "Cappuccino Mousse",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "34",
      "image": "/images/desserts/caramel.png",
      "name": "Caramel Flan",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "35",
      "image": "/images/desserts/eclair.png",
      "name": "Chocolate Eclair",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "36",
      "image": "/images/desserts/grandmarnier.png",
      "name": "Chocolate Grand Marnier",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "37",
      "image": "/images/desserts/mix_fruit_florentine.png",
      "name": "Mixed Fruit Florentine",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "38",
      "image": "/images/desserts/black_forest.png",
      "name": "Black Forest",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "39",
      "image": "/images/desserts/cappuccino_eclair.png",
      "name": "Cappuccino Eclair",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "40",
      "image": "/images/desserts/carrot_cake.png",
      "name": "Carrot Cake",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "41",
      "image": "/images/desserts/chocolate_suicide.png",
      "name": "Chocolate Suicide",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
    Object {
      "category": "",
      "countInStock": 20,
      "description": "",
      "id": "42",
      "image": "/images/desserts/mango_mousse.png",
      "name": "Mango Mousse",
      "numReviews": 10,
      "price": 3,
      "rating": 4.5,
    },
  ],
}
`;
